File: dagger/app/src/main/java/com/dagger/mvp/widget/MVPActivity.java
package com.dagger.mvp.widget;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.widget.Toast;

import com.dagger.BaseActivity;
import com.dagger.R;
import com.dagger.application.App;
import com.dagger.mvp.model.MVPBean;
import com.dagger.mvp.model.RegisterModel;
import com.dagger.mvp.presenter.MVPPresenterImpl;
import com.dagger.mvp.view.DaggerMVPComponent;
import com.dagger.mvp.view.MVPComponent;
import com.dagger.mvp.view.MVPView;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import io.reactivex.network.manager.RxNetWork;

/**
 * by y on 2017/5/31.
 */

public class MVPActivity extends BaseActivity implements MVPView, SwipeRefreshLayout.OnRefreshListener {

    private SwipeRefreshLayout swipeRefreshLayout;
    private MVPAdapter adapter;

    @Inject
    MVPPresenterImpl presenter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_simple);
        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.recyclerView);
        swipeRefreshLayout = (SwipeRefreshLayout) findViewById(R.id.refreshLayout);

        swipeRefreshLayout.setEnabled(false);
        swipeRefreshLayout.setOnRefreshListener(this);
        swipeRefreshLayout.post(new Runnable() {
            @Override
            public void run() {
                onRefresh();
            }
        });

        adapter = new MVPAdapter(new ArrayList<MVPBean>());
        recyclerView.setHasFixedSize(true);
        recyclerView.setLayoutManager(new LinearLayoutManager(recyclerView.getContext()));
        recyclerView.setAdapter(adapter);


        MVPComponent build = DaggerMVPComponent.builder()
                .registerModel(new RegisterModel(this))
                .applicationComponent(App.getBuild((App) getApplicationContext()))
                .build();
        build.register(this);

        App application = build.getApplication();
    }

    @Override
    public void onRefresh() {
        presenter.startNetWork();
    }

    @Override
    public void showProgress() {
        swipeRefreshLayout.setRefreshing(true);
    }

    @Override
    public void hideProgress() {
        swipeRefreshLayout.setRefreshing(false);
    }

    @Override
    public void onNetError() {
        Toast.makeText(getApplicationContext(), "error", Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onNetSuccess(List<MVPBean> list) {
        adapter.addAll(list);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        RxNetWork.getInstance().cancel(RxNetWork.TAG);
    }
}

File: UtilsSample/Dagger/src/main/java/sample/util/develop/android/dagger/mvp/widget/MVPActivity.kt
package sample.util.develop.android.dagger.mvp.widget

import android.os.Bundle
import android.support.v4.widget.SwipeRefreshLayout
import android.support.v7.widget.LinearLayoutManager
import android.widget.Toast
import io.reactivex.network.RxNetWork
import kotlinx.android.synthetic.main.dagger_activity_simple.*
import sample.util.develop.android.dagger.BaseActivity
import sample.util.develop.android.dagger.R
import sample.util.develop.android.dagger.application.App
import sample.util.develop.android.dagger.mvp.model.MVPBean
import sample.util.develop.android.dagger.mvp.model.RegisterModel
import sample.util.develop.android.dagger.mvp.presenter.MVPPresenterImpl
import sample.util.develop.android.dagger.mvp.view.DaggerMVPComponent
import sample.util.develop.android.dagger.mvp.view.MVPView
import java.util.*
import javax.inject.Inject

/**
 * by y on 2017/5/31.
 */

class MVPActivity : BaseActivity(), MVPView, SwipeRefreshLayout.OnRefreshListener {

    private lateinit var adapter: MVPAdapter

    @Inject
    lateinit var presenter: MVPPresenterImpl

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.dagger_activity_simple)

        refreshLayout.isEnabled = false
        refreshLayout.setOnRefreshListener(this)
        refreshLayout.post { onRefresh() }

        adapter = MVPAdapter(ArrayList())
        recyclerView.setHasFixedSize(true)
        recyclerView.layoutManager = LinearLayoutManager(recyclerView.context)
        recyclerView.adapter = adapter

        val build = DaggerMVPComponent.builder()
            .registerModel(RegisterModel(this))
            .applicationComponent(App.getBuild(applicationContext as App))
            .build()
        build.register(this)
        val application = build.application
    }

    override fun onRefresh() {
        presenter.startNetWork()
    }

    override fun showProgress() {
        refreshLayout.isRefreshing = true
    }

    override fun hideProgress() {
        refreshLayout.isRefreshing = false
    }

    override fun onNetError() {
        Toast.makeText(applicationContext, "error", Toast.LENGTH_SHORT).show()
    }

    override fun onNetSuccess(list: List<MVPBean>) {
        adapter.addAll(list)
    }

    override fun onDestroy() {
        super.onDestroy()
        RxNetWork.instance.cancelDefaultKey()
    }
}
